frontend/document/koptinterface.lua

```diff
@@ -177,6 +177,11 @@ function KoptInterface:getContextHash(doc, pageno, bbox, hash_list)
     table.insert(hash_list, bbox.y1)
     table.insert(hash_list, canvas_size.w)
     table.insert(hash_list, canvas_size.h)
+    -- NOTE: only add an entry for the rendering mode if colored output is
+    -- enabled to avoid invalidating previously cached data on B&W devices.
+    if doc.render_color then
+        table.insert(hash_list, 'color')
+    end
```

Comment:
Is that such a big deal? They'd probably be kicked out of the cache soon anyway; it's pretty much just the first new page or two when reopening after an update.

Reply:
OK, I'll add it unconditionally then.